proposals/readonly-struct-init.md

```diff
@@ -0,0 +1,161 @@
+# readonly object initializers 
+
+* [x] Proposed
+* [ ] Prototype: Not Started
+* [ ] Implementation: Not Started
+* [ ] Specification: Below
+
+## Summary
+
+This feature will allow for assigning of `readonly` fields and get only auto-implemented properties
```

Comment:
Would it make sense to limit this to just assigning of `readonly` fields, and omit properties ?

You cannot freely convert back and forth between auto-implemented property with this. It makes the extra ceremony required by properties pointless and just unnecessary overhead.

Reply:
We may also need to review the framework and change auto-implemented properties affected by this to explicitly implemented properties. We do not necessary want to promise that all properties that happen to be implemented as auto-implemented properties today are going to be like that forever.

Reply:
Note that you can't freely convert between auto-properties and regular properties today. There are situations where it causes breaking changes at the source level. This is indeed a new case though.

Reply:
> you can't freely convert between auto-properties and regular properties today

Does it cause breaking changes in the public surface, or just in other places within the same compilation unit?

We have been freely converting between auto-properties and regular properties in the framework implementation a lot, and I do not remember ever encountering a case where it would cause a breaking change.

Reply:
It can cause a break in the public surface area when it changes the shape of the backing fields. Consider:

``` csharp
struct S { 
  public int Value => 42; 
}
``` 

In version 2 if you change it to 

``` csharp
struct S {
  public int Value { get; } = 42;
}
```

This breaks any code that was doing member wise initialization of `S`.